### DISCUSSION TITLE
Porting more Processing tutorials for py5 (via p5 and Processing.py)

### DISCUSSION THREAD

**Discussion by villares** [py5coding/py5generator#333] [Ideas]
Our friendly neighbors from the [p5](https://p5.readthedocs.io/) (a.k.a. [p5py](https://github.com/p5py/p5)) project have done some wonderful work porting Processing Java tutorials, some of which  have never before been ported to Processing Python mode.

##  [P5 Docs » Tutorials](https://p5.readthedocs.io/en/latest/tutorials/index.html )
- [Coordinate System and Shapes](https://p5.readthedocs.io/en/latest/tutorials/coordinate%20system%20and%20shapes.html)
- [Color](https://p5.readthedocs.io/en/latest/tutorials/color.html)
- [Objects](https://p5.readthedocs.io/en/latest/tutorials/objects.html)
- [**Interactivity**](https://p5.readthedocs.io/en/latest/tutorials/interactivity.html)
- [Typography](https://p5.readthedocs.io/en/latest/tutorials/typography.html)
- [Strings and Drawing Text](https://p5.readthedocs.io/en/latest/tutorials/strings%20and%20drawing%20text.html)
- [Arrays](https://p5.readthedocs.io/en/latest/tutorials/arrays.html)
- [Images and Pixels](https://p5.readthedocs.io/en/latest/tutorials/images%20and%20pixels.html)
- [**Curves**](https://p5.readthedocs.io/en/latest/tutorials/curves.html)
- [2D Transformations](https://p5.readthedocs.io/en/latest/tutorials/2D%20transformations.html)
- [**PShape**](https://p5.readthedocs.io/en/latest/tutorials/pshape.html)
- [**Data**](https://p5.readthedocs.io/en/latest/tutorials/data.html)
- [**Trigonometry** Primer](https://p5.readthedocs.io/en/latest/tutorials/trigonometry%20primer.html)
- [Two-dimensional Arrays](https://p5.readthedocs.io/en/latest/tutorials/two%20dimensional%20arrays.html)
- [**Electronics**](https://p5.readthedocs.io/en/latest/tutorials/electronics.html)
- [**Network**](https://p5.readthedocs.io/en/latest/tutorials/network.html)
- [**Vector**](https://p5.readthedocs.io/en/latest/tutorials/vector.html)

## [Processing Python Mode Tutorials](https://py.processing.org/tutorials/)

- [**Getting Started**](https://py.processing.org/tutorials/gettingstarted) by Casey Reas and Ben Fry. Level: Beginner
- [**Processing Overview**](https://py.processing.org/tutorials/overview) by Ben Fry a d Casey Reas. A more detailed introduction to the different features of Processing. Level: Beginner
- [Coordinate System and Shapes](https://py.processing.org/tutorials/drawing) by Daniel Shiffman. Drawing simple shapes and using the coordinate system. Level: Beginner
- [Color](https://py.processing.org/tutorials/color) by Daniel Shiffman. An introduction to digital color. Level: Beginner
- [Interactivity](https://py.processing.org/tutorials/interactivity) by Casey Reas and Ben Fry. Introduction to Interactivity with the Mouse and Keyboard. Level: Beginner
- [Objects](https://py.processing.org/tutorials/objects) by Daniel Shiffman. The basics of object-oriented programming. Level: Beginner
- [Two-Dimensional Lists](https://py.processing.org/tutorials/2dlists) by Dan Shiffman. How to store data in a matrix using a two-dimensional list. Level: Intermediate
- [Images and Pixels](https://py.processing.org/tutorials/pixels) by Daniel Shiffman. How to load and display images as well as access their pixels. Level: Intermediate
- [Strings and Drawing Text](https://py.processing.org/tutorials/text) by Daniel Shiffman. Learn how to use the string class and display text onscreen. Level: Intermediate
- [2D Transformations](https://py.processing.org/tutorials/transform2d) by J David Eisenberg. Learn how to translate, rotate and scale shapes using 2D transformations. Level: Intermediate
- [**P3D**](https://py.processing.org/tutorials/p3d) by Daniel Shiffman. Developing advanced graphics applications in Processing using P3D (OpenGL) mode. Level: Advanced
- [**Anatomy of a Program**](https://py.processing.org/tutorials/anatomy) by J David Eisenberg. How do you analyze a problem and break it down into steps that the computer can do? Level: Advanced 

## Call to action!

I think we could now port them to py5,  with  credits to @arihantparsoya Arihant Parsoya, @abhikpal Abhik Pal and the other great p5 contributors... It doesn't seem hard (specially thanks to their efforts), but **we need volunteers**!

I believe this could be a great way for more people to get involved in contributing with py5 and a great resource for people learning it.

**Comment by hx2A** [comment]
This is a great idea! Perhaps we can try a few to see what kinds of changes need to be made? Some of the work can probably be automated.
